apps/builder/app/builder/features/pages/page-settings.tsx

```diff
@@ -1642,7 +1642,16 @@ export const PageSettings = ({
       onDuplicate={() => {
         const newPageId = duplicatePage(pageId);
         if (newPageId !== undefined) {
-          onDuplicate(newPageId);
+          // In `canvas.tsx`, within `subscribeStyles`, we use `requestAnimationFrame` (RAF) for style recalculation.
+          // After `duplicatePage`, styles are not yet recalculated.
+          // To ensure they are properly updated, we use double RAF.
+          requestAnimationFrame(() => {
```

Comment:
OMG, maybe there is some way to subscribe to some event or store value instead?

Reply:
In no way it should be solved here. Too much